modules/katamari/src/main/ts/ephox/katamari/api/Type.ts

```diff
@@ -1,35 +1,43 @@
-const typeOf = function (x: any) {
-  if (x === null) {
-    return 'null';
-  }
-  const t = typeof x;
-  if (t === 'object' && (Array.prototype.isPrototypeOf(x) || x.constructor && x.constructor.name === 'Array')) {
-    return 'array';
-  }
-  if (t === 'object' && (String.prototype.isPrototypeOf(x) || x.constructor && x.constructor.name === 'String')) {
-    return 'string';
-  }
-  return t;
-};
+const typeOf = (x: any): string =>
+  x === null ? 'null' : isArray(x) ? 'array' : isString(x) ? 'string' : typeof x;
 
-const isType = function (type: string) {
-  return function (value: any) {
-    return typeOf(value) === type;
-  };
-};
+const isType = <Yolo>(type: string) => (value: any): value is Yolo =>
```

Comment:
Why Yolo? Yolo as in https://en.m.wikipedia.org/wiki/Object_detection?

Reply:
`Yolo` as in "You only live once".  I'm indicating that this particular type is a lie.

Reply:
What do you think of something crazy like this?

```const isPrototype = <T> (pt: any, ptName: string) => (x: any): x is T =>
  x !== null && typeof x === 'object' && (pt.prototype.isPrototypeOf(x) || x.constructor && x.constructor.name === ptName);

export const isString: (x: any) => x is string = isPrototype(String, 'String');

export const isArray: (x: any) => x is Array<unknown> = isPrototype(Array, 'Array');
```

Reply:
Hmmm. I don't think that'll work.

Reply:
Seems to work with one minor addition to the `isString` check? I had to add change it to this:
```
const isPrototype = <T> (pt: any, ptName: string) => (x: any): x is T =>
  x !== null && typeof x === 'object' && (pt.prototype.isPrototypeOf(x) || x.constructor && x.constructor.name === ptName);

export const isString: (x: any) => x is string = isSimpleType('string') || isPrototype(String, 'String');

export const isArray: (x: any) => x is Array<unknown> = isPrototype(Array, 'Array');
```

as the prototype match only works if you create a string like so: `String('blah')`

Reply:
Tests were failing on IE. I've taken `isArray`, `isObject`, `isString` and  `typeOf` back to their original versions. Still a zip reduction of 1119 bytes.